src/sol_binder/nonce/file.py

```diff
@@ -27,10 +28,20 @@ def create(cls, w3: Web3, project_root: str,
     def name(cls):
         return "file"
 
-    def _get_and_increment(self, account: HexAddress) -> Optional[int]:
+    def _lock_file_path(self):
```

Comment:
Use the lockfile package which probably goes over all the best file-lock practices